### PR TITLE
[PF-387] Explicitly track Sam workspace resource ownership (via workspace stage)

### PR DESCRIPTION
**Pull request by zloery**
(This is a rewrite of #225 using the existing workspace stage instead of a new field)

This change alters WSM's behavior for workspaces owned by Rawls (indicated by workspace stage RAWLS_WORKSPACE, which is the default) such that WSM will no longer create or delete Sam resources for these workspacses.

Concretely, this means:
-WSM will not create a Sam resource when a user creates a RAWLS_WORKSPACE stage workspace. It will instead check that the user has access to an existing Sam workspace resource.
-WSM will not delete a Sam resource when a user deletes a RAWLS_WORKSPACE stage workspace. As before, deleting a WSM workspace still requires having delete permission on that workspace resource in Sam.
-Tests now explicitly use MC_WORKSPACE stage instead of the default RAWLS_WORKSPACE, as we do not set up Sam resources externally for them.

With this change, we should also update Rawls to call WSM's delete before calling Sam. In the current state, we're leaving behind zombie rows in WSM through the following process:

1. Rawls deletes Sam resource successfully.
2. Rawls calls WSM's delete endpoint with a user's token.
3. WSM sees that the workspace exists in its database, but will always get "false" when it asks Sam if the user has permission to delete. WSM does not delete the row from its database and returns a 403. These workspaces cannot be deleted by any user.
4. Rawls logs this as a warning and otherwise ignores it.

### CODE REVIEW

**Comment by davidangb** [comment]
fyi AS-666 (somehow appropriate) for zombie rows

**Review by davidangb — COMMENTED** [review]
fairly minor comments inline about code syntax. At the design level ... I wonder if it would be better to move the `isSamResourceOwner` logic up to the `WorkspaceCreateFlight` and `WorkspaceDeleteFlight` level, and simply not add the authz steps to those flights if the workspace is a RAWLS_WORKSPACE.

File: src/main/java/bio/terra/workspace/service/workspace/flight/CreateWorkspaceAuthzStep.java [hunk 1]
```diff
@@ -30,29 +32,53 @@ public CreateWorkspaceAuthzStep(SamService samService, AuthenticatedUserRequest
...
+      if (canReadExistingWorkspace(workspaceID)) {
+        return StepResult.getStepResultSuccess();
+      } else {
+        throw new WorkspaceNotFoundException(
+            String.format(
+                "Could not find pre-existing Sam resource for workspace %s. WSM will not create Sam resources for RAWLS_WORKSPACE stage workspaces.",
+                workspaceID));
+      }
+    }
+  }
```
[the remark this answers is not in the document]
Comment (zloery) [reply src/main/java/bio/terra/workspace/service/workspace/flight/CreateWorkspaceAuthzStep.java]: I think either way ends up pretty hard to parse, but I went with your suggestion of pulling this out to the flight level which simplifies things inside the steps

**Review by ddietterich — APPROVED** [review]
lgtm

File: src/main/java/bio/terra/workspace/service/workspace/flight/CreateWorkspaceAuthzStep.java [hunk 2]
```diff
@@ -30,19 +30,10 @@ public CreateWorkspaceAuthzStep(SamService samService, AuthenticatedUserRequest
...
-    // failing) to create
-    // the Sam resource. If the user doesn't have access, we'll default to the existing behavior and
-    // return the following
-    // error or success message from Sam.
-    if (!samService.isAuthorized(
-        userReq.getRequiredToken(),
-        SamConstants.SAM_WORKSPACE_RESOURCE,
-        workspaceID.toString(),
-        SamConstants.SAM_WORKSPACE_READ_ACTION)) {
+    // Even though WSM should own this resource, Stairway steps can run multiple times, so it's
```
[the remark this answers is not in the document]
Comment (zloery) [reply src/main/java/bio/terra/workspace/service/workspace/flight/CreateWorkspaceAuthzStep.java:37]: yep, exactly. added a class-level comment explaining this
